model/src/payouts/quanto.rs

```diff
@@ -0,0 +1,609 @@
+
+        match self.fee_offset {
+            CompleteFee::LongPaysShort(long_owes) if long_owes > initial_margin_long => {
+                return Err(Error::LongOwesTooMuch {
+                    owing: long_owes,
+                    margin: initial_margin_long,
+                });
+            }
+            CompleteFee::ShortPaysLong(short_owes) if short_owes > initial_margin_short => {
+                return Err(Error::ShortOwesTooMuch {
+                    owing: short_owes,
+                    margin: initial_margin_short,
+                });
+            }
+            _ => {}
+        }
+
+        let long_liquidation_interval = self
+            .long_liquidation_interval()
+            .context("Could not calculate long liquidation interval")?;
+        let short_liquidation_interval = self
+            .short_liquidation_interval()
+            .context("Could not calculate short liquidation interval")?;
+
+        let long_liquidation_threshold = long_liquidation_interval.end();
+        let short_liquidation_threshold = short_liquidation_interval.start();
+
+        let long_liquidation_payout = Payout {
+            interval: long_liquidation_interval.clone(),
+            long: Amount::ZERO,
+            short: initial_margin_total,
+        };
+
+        let mut payouts = vec![long_liquidation_payout];
+
+        // We want to split the settlement region of the curve into `n-payouts - 2` segments
+        let step = Decimal::from(short_liquidation_threshold - long_liquidation_threshold)
+            / Decimal::from(self.n_payouts - 2);
+        let step = step.to_u64().context("Could not convert step to u64")?;
+
+        // We start building the settlement intervals right after the end of the long liquidation
+        // interval
+        let mut prev_upper_bound = *long_liquidation_threshold;
+        loop {
+            let lower_bound = prev_upper_bound + 1;
+            let upper_bound = lower_bound + step;
+
+            // The settlement region of the curve ends as soon as the sub-interval we are building
+            // either touches or overlaps with the start of the short liquidation interval
+            if upper_bound >= short_liquidation_threshold - 1 {
+                break;
+            }
+
+            let interval = lower_bound..=upper_bound;
+            let payout = self
+                .payout_at_interval(interval.clone())
+                .with_context(|| format!("Could not calculate payout at interval {interval:?}"))?;
+            payouts.push(payout);
+
+            prev_upper_bound = upper_bound;
+        }
+
+        // We have to consider a special case if the last settlement interval is smaller than every
+        // other settlement interval
+        if prev_upper_bound + 1 < *short_liquidation_threshold {
+            let lower_bound = prev_upper_bound + 1;
+            let upper_bound = short_liquidation_threshold - 1;
+
+            let interval = lower_bound..=upper_bound;
```

Comment:
is `10..10` a valid interval? 

You could get this if: 

`prev_upper_bound=9`, `short_liquidation_threshold=11`.

Reply:
Yep, single-valued intervals are valid and we have to support them.

Normally they will only occur for the special case in the settlement interval. But it could be that we model every price value as a separate CET with sufficient resolution!